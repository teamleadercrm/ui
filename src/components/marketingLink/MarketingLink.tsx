import React, { forwardRef, useRef, useImperativeHandle, ReactNode } from 'react';
import cx from 'classnames';
import Box from '../box';
import theme from './theme.css';
import uiUtilities from '@teamleader/ui-utilities';
import { GenericComponent } from '../../@types/types';
import { BoxProps } from '../box/Box';

export interface MarketingLinkProps extends Omit<BoxProps, 'ref'> {
  /** The content to display inside the link. */
  children: ReactNode;
  /** A class name for the link to give custom styles. */
  className?: string;
  /** A custom element to be rendered */
  element?: React.ElementType;
  /** Callback function that is fired when mouse leaves the component. */
  onMouseLeave?: (event: React.MouseEvent) => void;
  /** Callback function that is fired when the mouse button is released. */
  onMouseUp?: (event: React.MouseEvent) => void;
}

const MarketingLink: GenericComponent<MarketingLinkProps> = forwardRef<HTMLElement, MarketingLinkProps>(
  ({ children, className = '', element = 'a', ...others }, ref) => {
    const linkRef = useRef<HTMLElement>(null);
    useImperativeHandle<HTMLElement | null, HTMLElement | null>(ref, () => linkRef.current);

    const classNames = cx(uiUtilities['reset-font-smoothing'], theme['link'], className);

    return (
      <Box element={element} ref={linkRef} className={classNames} data-teamleader-ui="marketing-link" {...others}>
        {children}
      </Box>
    );
  },
);

MarketingLink.displayName = 'MarketingLink';

export default MarketingLink;
