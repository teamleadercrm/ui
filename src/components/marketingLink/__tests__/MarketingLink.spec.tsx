import React from 'react';
import { render } from '@testing-library/react';
import MarketingLink from '..';

describe('Component - MarketingLink', () => {
  it('renders', async () => {
    const { asFragment } = render(<MarketingLink href="https://foobar.com">Foobar</MarketingLink>);
    expect(asFragment()).toMatchSnapshot();
  });
});
