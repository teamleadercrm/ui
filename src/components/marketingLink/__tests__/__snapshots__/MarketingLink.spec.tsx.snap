// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - MarketingLink renders 1`] = `
<DocumentFragment>
  <a
    class="box link"
    data-teamleader-ui="marketing-link"
    href="https://foobar.com"
  >
    Foobar
  </a>
</DocumentFragment>
`;
